### PR TITLE
Arbitrarily color boxes / junctions of a wiring diagram

### PR DESCRIPTION
**Pull request by kris-brown**
This is one way we could allow users to color wiring diagrams. A dictionary mapping box IDs to strings (e.g. `"lightblue"`) can be passed to `to_graphviz`, with the default behavior being unchanged (unfilled boxes, black junctions).

### CODE REVIEW

**Comment by jpfairbanks** [comment]
It looks like #558 could use this too.

File: src/graphics/GraphvizWiringDiagrams.jl [hunk 1]
```diff
@@ -84,6 +84,7 @@ function to_graphviz(f::WiringDiagram;
     port_size::String="24", junction_size::String="0.05",
     outer_ports::Bool=true, anchor_outer_ports::Bool=true,
     graph_attrs::AbstractDict=Dict(), node_attrs::AbstractDict=Dict(),
+    node_colors::Dict{Int,String}=Dict{Int,String}(),
```
Comment (epatters) [inline src/graphics/GraphvizWiringDiagrams.jl:87]: Would replace with ``node_colors::AbstractDict=Dict()`` to avoid unnecessarily strong typing.

File: src/graphics/GraphvizWiringDiagrams.jl [hunk 2]
```diff
@@ -219,7 +226,7 @@ function box_html_label(nin::Int, nout::Int, text_label::String;
     Graphviz.Html("""
       <TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0">
       <TR><TD>$(orientation == TopToBottom ? input_label : output_label)</TD></TR>
-      <TR><TD $(html_attributes(attrs))>$(escape_html(text_label))</TD></TR>
+      <TR><TD $(html_attributes(attrs))>$((text_label))</TD></TR>
```
Comment (epatters) [inline src/graphics/GraphvizWiringDiagrams.jl:229]: This change shouldn't be in this PR, as it addresses a separate issue and also I suspect it needs to be handled more carefully.

File: src/graphics/GraphvizWiringDiagrams.jl [hunk 3]
```diff
@@ -481,7 +489,7 @@ end
 
 """ Create a label for the main content of a box.
 """
-node_label(box_value) = box_label(MIME("text/plain"), box_value)
+node_label(box_value) = box_label(MIME("text/html"), box_value)
```
Comment (epatters) [inline src/graphics/GraphvizWiringDiagrams.jl:492]: Likewise for this change.

**Review by epatters — CHANGES_REQUESTED** [review]

File: src/graphics/GraphvizWiringDiagrams.jl (hunk 2 above)
Comment (kris-brown) [reply src/graphics/GraphvizWiringDiagrams.jl:229]: sorry - that hack wasn't supposed to leak into the PR!

File: src/graphics/GraphvizWiringDiagrams.jl [hunk 4]
```diff
@@ -416,6 +423,7 @@ function to_graphviz_property_graph(d::UndirectedWiringDiagram;
   # Create nodes for boxes.
   box_vs = add_vertices!(graph, nboxes(d))
   set_vprop!(graph, box_vs, :id, [ "box$b" for b in boxes(d) ])
+  set_vprop!(graph, box_vs, :fillcolor, [ "green" for b in boxes(d) ])
```
Comment (epatters) [inline src/graphics/GraphvizWiringDiagrams.jl:426]: Sorry, one more thing. Was this change intended? I feel like the UWD boxes probably shouldn't have a color by default, but if they are colored, that color should at least be configurable.

**Review by epatters — COMMENTED** [review]
Besides the question/comment below, it looks like you'll need to rebase due to conflict with your other PR, now merged.

File: src/graphics/GraphvizWiringDiagrams.jl (hunk 4 above)
Comment (kris-brown) [reply src/graphics/GraphvizWiringDiagrams.jl]: definitely unintended, thanks!

**Review by epatters — APPROVED** [review]
